### PR TITLE
Add a page for search_traces

### PR DESCRIPTION
**Pull request by TomeHirata**
### Related Issues/PRs

N/A

### What changes are proposed in this pull request?

This PR create a new page for the usage of the mlflow.search_traces and MlflowClient.search_traces API.

### How is this PR tested?

- [x] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests

### Does this PR require documentation update?

- [ ] No. You can skip the rest of this section.
- [x] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [x] Instructions

#### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [x] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

#### Should this PR be included in the next patch release?

`Yes` should be selected for bug fixes, documentation updates, and other small changes. `No` should be selected for new features and larger changes. If you're unsure about the release classification of this PR, leave this unchecked to let the maintainers decide.

<details>
<summary>What is a minor/patch release?</summary>

- Minor release: a release that increments the second part of the version number (e.g., 1.2.0 -> 1.3.0).
  Bug fixes, doc updates and new features usually go into minor releases.
- Patch release: a release that increments the third part of the version number (e.g., 1.2.0 -> 1.2.1).
  Bug fixes and doc updates usually go into patch releases.

</details>

- [ ] Yes (this PR will be cherry-picked and included in the next patch release)
- [x] No (this PR will be included in the next minor release)

### CODE REVIEW

File: docs/source/llms/tracing/search-traces.rst [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+        order_by=["timestamp DESC"],
+    )
+    # Returns Trace #3, Trace #2, Trace #1
+
+.. _extract_fields:
+
+Extract Specific Fields
+-----------------------
+
+In addition to the search functionalities mentioned above, the fluent API :py:func:`mlflow.search_traces` enables you
```
Comment (B-Step62) [inline docs/source/llms/tracing/search-traces.rst:190]:
```suggestion
In addition to the search functionalities mentioned above, the fluent API :py:func:`mlflow.search_traces` allows you 
```
nit

File: docs/source/llms/tracing/search-traces.rst [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+.. code-block:: python
+
+    traces = mlflow.search_traces(
+        extract_fields=["morning_greeting.inputs", "morning_greeting.outputs"],
+        experiment_ids=[morning_experiment.experiment_id],
+    )
+
+    print(traces)
+
+Output:
```
Comment (B-Step62) [inline docs/source/llms/tracing/search-traces.rst:204]:
```suggestion
The output Pandas DataFrame will contain the additional columns for the extracted span fields:
```

File: docs/source/llms/tracing/search-traces.rst [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+
+.. code-block:: text
+
+        request_id                              ...     morning_greeting.inputs        morning_greeting.outputs
+    0   053adf2f5f5e4ad68d432e06e254c8a4        ...     {'name': 'Tom'}                'Good morning Tom.'
+
+Lastly, you can convert the pandas DataFrame to the MLflow LLM evaluation dataset format and evaluate your language model.
```
Comment (B-Step62) [inline docs/source/llms/tracing/search-traces.rst:211]: Thank you for adding this example!

File: docs/source/llms/tracing/search-traces.rst [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+
+Filter Traces by Name
+^^^^^^^^^^^^^^^^^^^^^
+
+Search for traces by the ``attributes.name`` keyword:
+
+.. code-block:: python
+
+    client.search_traces(
+        experiment_ids=[morning_experiment.experiment_id, evening_experiment.experiment_id],
```
Comment (B-Step62) [inline docs/source/llms/tracing/search-traces.rst:114]:
```suggestion
        experiment_ids=all_experiment_ids,
```
nit: I'd define a variable to reduce code unrelated to the main topic.

File: docs/source/llms/tracing/search-traces.rst [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+
+        request_id                              ...     morning_greeting.inputs        morning_greeting.outputs
+    0   053adf2f5f5e4ad68d432e06e254c8a4        ...     {'name': 'Tom'}                'Good morning Tom.'
+
+Lastly, you can convert the pandas DataFrame to the MLflow LLM evaluation dataset format and evaluate your language model.
+
+.. code-block:: python
+    
+    import pandas as pd
+
```
Comment (B-Step62) [inline docs/source/llms/tracing/search-traces.rst:216]:
```suggestion
```

looks like not used?

File: docs/source/llms/tracing/search-traces.rst [hunk 6]
```diff
@@ -0,0 +1,228 @@
...
+    is not provided.
+
+.. code-block:: python
+
+    from mlflow import MlflowClient
+
+    client = MlflowClient()
+
+    client.search_traces(experiment_ids=[morning_experiment.experiment_id])
+    # Returns Trace #1
```
[the remark this answers is not in the document]
Comment (TomeHirata) [reply docs/source/llms/tracing/search-traces.rst:95]: Sure, I guess putting screenshots every time is probably noisy, let me add some ascii to make it different.

File: docs/source/llms/tracing/search-traces.rst [hunk 7]
```diff
@@ -0,0 +1,228 @@
...
+
+    mlflow.search_traces(experiment_ids=[morning_experiment.experiment_id])
+    # Returns Trace #1
+
+Search Traces with **filter_string**
+------------------------------------
+
+The ``filter_string`` argument provides a flexible way to query traces using a **Domain-Specific Language (DSL)**, 
+which is inspired by SQL. The DSL supports various attributes and allows for combining multiple conditions.
```
[the remark this answers is not in the document]
Comment (TomeHirata) [reply docs/source/llms/tracing/search-traces.rst:104]: It is how we explain filter_string for tracking, so I think it is ok to use a terminology here.

[the remark this answers is not in the document]
Comment (B-Step62) [reply docs/source/llms/tracing/search-traces.rst]: I don't have strong preference here, but I would avoid jargons as much as possible. Our audience is not necessarily an engineer but also come from academic background.

**Comment by TomeHirata** [comment]
@B-Step62 Thank you for the review. I've addressed your comments, can you take another look?

File: docs/source/llms/tracing/search-traces.rst [hunk 8]
```diff
@@ -0,0 +1,245 @@
...
+:py:meth:`MlflowClient.search_traces() <mlflow.client.MlflowClient.search_traces>` and :py:func:`mlflow.search_traces`.
+
+- :py:meth:`MlflowClient.search_traces() <mlflow.client.MlflowClient.search_traces>`: This method allows you to filter traces using experiment IDs, 
+  filter strings, and other parameters.
+
+- :py:func:`mlflow.search_traces`: A higher-level fluent API that returns a pandas DataFrame, with each row representing 
+  a trace. It supports the same filtering capabilities as `MlflowClient.search_traces` and additionally allows you to specify 
+  fields to extract from traces. See :ref:`extract_fields` for details. 
+
+The pandas Dataframe returned by consists of the following columns by default:
```
Comment (B-Step62) [inline docs/source/llms/tracing/search-traces.rst:14]:
```suggestion
The pandas Dataframe returned by the :py:func:`mlflow.search_traces` consists of the following columns by default:
```

File: docs/source/llms/tracing/search-traces.rst [hunk 9]
```diff
@@ -0,0 +1,245 @@
...
+- spans: Spans in the trace.
+- tags: Tags associated with the trace.
+
+.. figure:: ../../_static/images/llms/tracing/search-traces.png
+    :alt: Search Traces Output
+    :width: 80%
+    :align: center
+
+Basic Usage of Search Traces
+----------------------------
```
Comment (B-Step62) [inline docs/source/llms/tracing/search-traces.rst:33]:
```suggestion
Basic Usage of Search Traces
----------------------------
```

File: docs/source/llms/tracing/search-traces.rst [hunk 10]
```diff
@@ -0,0 +1,245 @@
...
+        mlflow.update_current_trace(tags={"person": name})
+        return f"Good evening {name}."
+
+
+    @mlflow.trace(span_type=SpanType.TOOL)
+    def goodbye():
+        raise Exception("Cannot say goodbye")
+
+
+    # Execute methods with experiments
```
Comment (B-Step62) [inline docs/source/llms/tracing/search-traces.rst:64]:
```suggestion
    # Execute the methods within different experiments
```

File: docs/source/llms/tracing/search-traces.rst [hunk 11]
```diff
@@ -0,0 +1,245 @@
...
+   * - Evening Experiment
+     - ``evening_greeting``
+     - ``Mary``
+     - ``OK``
+   * - Evening Experiment
+     - ``goodbye``
+     - ``N/A``
+     - ``ERROR``
+
+Then, you can search traces by `experiment_ids` using either :py:func:`mlflow.search_traces` or
```
Comment (B-Step62) [inline docs/source/llms/tracing/search-traces.rst:98]:
```suggestion
Then, you can search traces by ``experiment_ids`` using either :py:func:`mlflow.search_traces` or 
```

**Review by B-Step62 — APPROVED** [review]
LGTM! Thanks for the addition!

File: docs/source/llms/tracing/search-traces.rst (hunk 8 above)
Comment (TomeHirata) [reply docs/source/llms/tracing/search-traces.rst:14]: Forgot to add ref here.. good catch!

**Pull request merged**